Mathlib/Algebra/Algebra/Subalgebra/Basic.lean

```diff
@@ -566,42 +552,41 @@ variable (p : Submodule R A)
 def toSubalgebra (p : Submodule R A) (h_one : (1 : A) ∈ p)
     (h_mul : ∀ x y, x ∈ p → y ∈ p → x * y ∈ p) : Subalgebra R A :=
   { p with
-    mul_mem' := h_mul
+    mul_mem' := fun hx hy ↦ h_mul _ _ hx hy
+    one_mem' := h_one
     algebraMap_mem' := fun r => by
       rw [Algebra.algebraMap_eq_smul_one]
       exact p.smul_mem _ h_one }
 #align submodule.to_subalgebra Submodule.toSubalgebra
 
 @[simp]
 theorem mem_toSubalgebra {p : Submodule R A} {h_one h_mul} {x} :
-    x ∈ p.toSubalgebra h_one h_mul ↔ x ∈ p :=
-  Iff.rfl
+    x ∈ p.toSubalgebra h_one h_mul ↔ x ∈ p := Iff.rfl
 #align submodule.mem_to_subalgebra Submodule.mem_toSubalgebra
 
 @[simp]
 theorem coe_toSubalgebra (p : Submodule R A) (h_one h_mul) :
-    (p.toSubalgebra h_one h_mul : Set A) = p :=
-  rfl
+    (p.toSubalgebra h_one h_mul : Set A) = p := rfl
 #align submodule.coe_to_subalgebra Submodule.coe_toSubalgebra
 
-@[simp]
-theorem toSubalgebra_mk (s : Set A) (h0 hadd hsmul h1 hmul) :
-    (Submodule.mk s hadd h0 hsmul : Submodule R A).toSubalgebra h1 hmul =
-      Subalgebra.mk s (@hmul) h1 (@hadd) h0 fun r =>
-        by
-        rw [Algebra.algebraMap_eq_smul_one]
-        exact hsmul r h1 :=
-  rfl
-#align submodule.to_subalgebra_mk Submodule.toSubalgebra_mk
+-- Porting note: not ported because we no longer have flat structures
+-- @[simp]
+-- theorem toSubalgebra_mk (s : Set A) (h0 hadd hsmul h1 hmul) :
+--     (Submodule.mk s hadd h0 hsmul : Submodule R A).toSubalgebra h1 hmul =
+--       Subalgebra.mk s (@hmul) h1 (@hadd) h0 fun r => by
+--         rw [Algebra.algebraMap_eq_smul_one]
+--         exact hsmul r h1 :=
+--   rfl
+-- #align submodule.to_subalgebra_mk Submodule.toSubalgebra_mk
```

Comment:
I think this should be a `#noalign`.

Although I'd argue we should still have this lemma, it will just need to be stated differently to match the new signature of `Subalgebra.mk`.

Reply:
Changed the statement, added porting note.